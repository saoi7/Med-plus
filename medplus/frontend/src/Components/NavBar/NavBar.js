import React from 'react';
import { Link, Switch, Route, BrowserRouter as Router } from 'react-router-dom';
import { HiHome } from 'react-icons/hi';

function NavBar() {
    return (
        <nav className="nav-bar">
            <Link to="/home" className="font-medium">
                Reminders
            </Link>
            <Link to="/home" className="font-medium">
                <HiHome />
            </Link>
            <Link to="/profile" className="font-medium">
                Profile
            </Link>
        </nav>
    );
}

export { NavBar };
