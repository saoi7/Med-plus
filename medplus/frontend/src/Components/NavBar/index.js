import React from 'react';
import { Link, Switch, Route, BrowserRouter as Router } from 'react-router-dom';
import { HiHome } from 'react-icons/hi';
import * as ROUTES from '../../constants/routes';
import { withFirebase } from '../Firebase';

function NavBar({ firebase }) {
    return (
        <nav className="nav-bar">
            <Link to={ROUTES.LANDING} onClick={firebase.doSignOut} className="font-medium">
                Sign Out
            </Link>
            <Link to={ROUTES.HOME} className="font-medium">
                <HiHome />
            </Link>
            <Link to={ROUTES.ACCOUNT} className="font-medium">
                Profile
            </Link>
        </nav>
    );
}

export default withFirebase(NavBar);
