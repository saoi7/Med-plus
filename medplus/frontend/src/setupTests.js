// jest-dom adds custom jest matchers for asserting on DOM nodes.
// allows you to do things like:
// expect(element).toHaveTextContent(/react/i)
// learn more: https://github.com/testing-library/jest-dom
import '@testing-library/jest-dom';
import { unmountComponentAtNode } from "react-dom";

let test_dom_container = null;
beforeEach(() => {
  // setup a DOM element as a render target
  test_dom_container = document.createElement("div");
  document.body.appendChild(test_dom_container);
});

afterEach(() => {
  // cleanup on exiting
  unmountComponentAtNode(test_dom_container);
  test_dom_container.remove();
  test_dom_container = null;
});

export { test_dom_container };
