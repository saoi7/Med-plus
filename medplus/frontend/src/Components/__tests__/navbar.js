import { MOCK_DB_MED_ENTRIES } from './mock_db_med_entries';
import {test_dom_container} from '../../setupTests';
import { render, unmountComponentAtNode } from "react-dom";
import { act } from "react-dom/test-utils";
import renderer from 'react-test-renderer';
import { getByTestId, fireEvent } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import { MOCK_FIREBASE_PROP } from './mock_firebase_props';
import NavBar from '../NavBar';
import flushPromises from 'flush-promises';

test("snapshot test of NavBar", () => {
    const COMPONENT= renderer.create(
        <MemoryRouter>
            <NavBar />
        </MemoryRouter>
    );
    let tree = COMPONENT.toJSON();
    expect(tree).toMatchSnapshot();
});

test("make sure NavBar is implemented with <nav> element", () => {
    act(() => {
        render(<MemoryRouter><NavBar /></MemoryRouter>, test_dom_container);
    });
    const NAV_ELEMENT = test_dom_container.childNodes[0];
    expect(NAV_ELEMENT.tagName).toBe('NAV');
});
