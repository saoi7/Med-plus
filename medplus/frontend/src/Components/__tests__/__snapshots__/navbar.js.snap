// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot test of NavBar 1`] = `
<nav
  className="nav-bar"
>
  <a
    className="font-medium"
    href="/home"
    onClick={[Function]}
  >
    Reminders
  </a>
  <a
    className="font-medium"
    href="/home"
    onClick={[Function]}
  >
    <svg
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 20 20"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M10.707 2.293a1 1 0 00-1.414 0l-7 7a1 1 0 001.414 1.414L4 10.414V17a1 1 0 001 1h2a1 1 0 001-1v-2a1 1 0 011-1h2a1 1 0 011 1v2a1 1 0 001 1h2a1 1 0 001-1v-6.586l.293.293a1 1 0 001.414-1.414l-7-7z"
      />
    </svg>
  </a>
  <a
    className="font-medium"
    href="/account"
    onClick={[Function]}
  >
    Profile
  </a>
</nav>
`;
